src/main.cpp

```diff
@@ -35,88 +37,89 @@ constexpr auto VERSION = "[Unkown version]";
+  tlg::info << "Touchégg is an app that runs in the background and transforms "
                "the gestures you make on your touchpad into visible actions in "
                "your desktop."
             << std::endl;
-  std::cout << "For more information please visit:" << std::endl;
-  std::cout << "https://github.com/JoseExposito/touchegg" << std::endl
+  tlg::info << "For more information please visit:" << std::endl;
+  tlg::info << "https://github.com/JoseExposito/touchegg" << std::endl
             << std::endl;
 
-  std::cout << "Option\t\tMeaning" << std::endl;
-  std::cout << " --daemon\tRun Touchégg in daemon mode. This mode starts a "
+  tlg::info << "Option\t\tMeaning" << std::endl;
+  tlg::info << " --daemon\tRun Touchégg in daemon mode. This mode starts a "
                "service that gathers gestures but executes no actions"
             << std::endl;
-  std::cout << " --client\tConnect to an existing Touchégg daemon and "
+  tlg::info << " --client\tConnect to an existing Touchégg daemon and "
                "execute actions in your desktop"
             << std::endl;
-  std::cout << "Without arguments Touchégg starts in client mode" << std::endl
+  tlg::info << "Without arguments Touchégg starts in client mode" << std::endl
             << std::endl;
 }
 
-int main(int argc, char **argv) {
-  printWelcomeMessage();
+int main(int argc, char** argv) {
+  ArgsParser args(argc, argv);
 
-  // Parse the command line arguments
-  bool daemonMode = false;
-  bool clientMode = (argc == 1);
-  double startThreshold = -1;
-  double finishThreshold = -1;
-
-  if (argc > 1) {
-    std::string param{argv[1]};  // NOLINT
-    daemonMode = (param == "--daemon");
-    clientMode = (param == "--client");
-
-    if (daemonMode && argc == 4) {
-      startThreshold = std::stod(argv[2]);   // NOLINT
-      finishThreshold = std::stod(argv[3]);  // NOLINT
-    }
-  }
+  // init Logger options
+  Logger::obj(args.debug, args.quiet);
 
-  if (!daemonMode && !clientMode) {
-    std::cout << "Invalid command line arguments" << std::endl;
+  // test log options
+  // tlg::error << "ERROR!!!" << std::endl;
+  // tlg::warning << "Warning!" << std::endl;
+  // tlg::info << "A very informative message." << std::endl;
+  // tlg::debug << "DBG: 0xdeadbeef" << std::endl;
+  // tlg::warning << std::endl << std::endl;
+
+  /*
+   * NOTE: Currently only appears if --debug option specified! You might want to
+   * change the LogLevel used in this func for msgs Jose.
+   */
+  printWelcomeMessage();
+
+  if (args.daemonMode == args.clientMode) {
+    tlg::error << "Invalid command line arguments" << std::endl;
     return -1;
   }
 
-  std::cout << "Starting Touchégg in "
-            << (daemonMode ? std::string{"daemon mode"}
-                           : std::string{"client mode"})
+  std::time_t t = std::time(NULL);
+  char mbstr[100];
+  std::strftime(mbstr, sizeof(mbstr) - 1, "[%F-%T%z] ", std::localtime(&t));
```

Comment:
I don't remember what we agreed on at the end, but if we are going to display a date, it should go in the logger, for every message.

Reply:
removed now!